### ISSUE TITLE
error: use of unstable library feature 'iter_arith'

### ISSUE THREAD

**Issue by jcoupey** [Tristramg/osm4routing2#5]
When running `cargo build`, I get the following:

```
   Compiling osm4routing v0.1.0 (file:///home/juliencoupey/Documents/repos/forks/osm4routing2)
src/osm4routing/models.rs:50:79: 50:84 error: use of unstable library feature 'iter_arith': bounds recently changed (see issue #27739)
src/osm4routing/models.rs:50         self.geometry.windows(2).map(|coords| distance(coords[0], coords[1])).sum()
                                                                                                           ^~~~~
```

The linked issue and the way it has been closed seems to point out the need to have a recent `rust` version. I've currently only installed the somewhat outdated `1.7.0` from the ubuntu repos.

Is there a simple workaround or do I need to install a recent `rust`? In which case it may be worth adding the required version in `readme`. ;-)

**Comment by Tristramg** [comment]
I suppose that indeed a newer version should work. Could you try to install a newer version?
With https://www.rustup.rs/ it’s quite easy. It might also fix the other issue you had.

(Thanks for reporting the problems)

**Comment by jcoupey** [comment]
Currently running through `rustup.sh` before piping to `sh`...

**Comment by jcoupey** [comment]
Thanks for the answer, solved by using version 1.13.0.
